src/main/java/com/qubit/solution/fenixedu/integration/cgd/services/studentPhoto/CgdStudentPhotoClient.java

```diff
@@ -0,0 +1,157 @@
+            }
+        }
+
+        public Person getPerson() {
+            Person personFromDocumentId = getPersonFromDocumentId();
+            if (isMemberNumberLikeIdDocumentNumber()) {
+                return personFromDocumentId;
+            } else if (isStudent()) {
+                Student student = Student.readStudentByNumber(getMemberAsNumber());
+                if (student != null) {
+                    return student.getPerson();
+                }
+            } else if (isStaff()) {
+                Optional<Employee> employeeOpt = Employee.findByNumber(this.memberNumber);
+                if (employeeOpt.isPresent()) {
+                    return employeeOpt.get().getPerson();
+                }
+                if (personFromDocumentId != null) {
+                    return personFromDocumentId;
+                }
+            }
+            return null;
+        }
+
+        public String getMemberNumber() {
+            return this.memberNumber;
+        }
+
+        public String getCategory() {
+            return category;
+        }
+    }
+
+    public static class PhotoOnCGD {
+        private byte[] photoContent;
+        private String fileName;
+
+        private PhotoOnCGD(byte[] ba, String fileName) {
+            this.photoContent = ba;
+            this.fileName = fileName;
+        }
+
+        public byte[] getPhotoContent() {
+            return photoContent;
+        }
+
+        public String getFileName() {
+            return fileName;
+        }
+
+    }
+
+    public Collection<MemberCGD> getStudentsWithPhotos() {
+        logger.debug("CgdPhotoWsClient.getStudentsWithPhotos()");
+        IStudentPhotoService ws = (IStudentPhotoService) getClient();
+        List<Member> members = ws.getAllStudentsByIES().getMember();
+
+        List<MemberCGD> memberCGDs = new ArrayList<MemberCGD>();
+        for (Member member : members) {
+            memberCGDs.add(new MemberCGD(member));
+        }
+        return memberCGDs;
+    }
+
+    public PhotoOnCGD getPhoto(final MemberCGD member) {
+        return getPhoto(member.getMemberNumber(), member.getCategory());
+    }
+
+    /*
+     * memberCategory parameter is optional. It is used by CGD to disambiguate between members with
+     * the same number and different category.
+     */
+    public PhotoOnCGD getPhoto(final String memberNumber, final String memberCategory) {
+        logger.debug("CgdPhotoWsClient.getStudentsWithPhotos()");
+        IStudentPhotoService ws = (IStudentPhotoService) getClient();
+        StudentPhoto studentPhoto = ws.getPhoto(memberNumber, memberCategory);
+        JAXBElement<byte[]> photoObj = studentPhoto.getPhoto();
+        byte[] photoContent = photoObj.getValue();
+        if (photoContent.length > 0) {
+            JAXBElement<String> fileNameJB = studentPhoto.getFileName();
+            String fileName = fileNameJB.getValue();
+            return new PhotoOnCGD(photoContent, fileName);
```

Comment:
@pmmsa The write transaction be done here. Otherwise you'll make people create the transaction outside of this method were a call to a web service will be made.
That may lead to performance issues, since if the web service degrade you'll start having long transations that make the FenixFramework lag.

Suggestion, do the transaction right here. You can extract the new into a new method and annotate it with a@atomic

```
@Atomic
private PhotoOnCgd createPhotoOnCGD(byte[] content, String filename) {
return new PhotoOnCGD(photoContent, fileName);
}

```

Reply:
Like we spoke, this makes no sense because PhotoOnCGD is a DTO and not a domain object. I'm invalidating thios request.